Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h

```diff
@@ -1879,8 +1913,10 @@ namespace internal {
     bool check_normals(const HalfedgeRange& hedges) const
     {
       std::size_t nb_patches = patch_id_to_index_map.size();
+      //std::vector<boost::optional<Vector_3> > normal_per_patch(nb_patches,boost::none);
```

Comment:
remove?

Reply:
Yes. I later replaced it with two vectors.   By the way I am not sure if `vector<bool>` is a good idea, and if it should not be `vector<int>`   instead.

Reply:
why would it be bad?

Reply:
Because it is more time consuming to change a bit inside an int.

Reply:
but you checked with optional and `vector<bool>` was faster? I agree with int since nb_patches is not expected to be very large

Reply:
I did not really see a difference in vtune but sticked to the second solution.  I don''t agree  on what you say on `nb_patches`.  If it is the number of patches of the data set and not only the patches incident to the vertex it may be arbitrarily  big.

Reply:
it is the total number of surface patches indeed, but still it's likely not to be arbitrarily big. A hundred would already be a strong data set...

Reply:
> Because it is more time consuming to change a bit inside an int.

That is very cheap, comparable to a memory load or a memory store from L1 cache. Both are about 3-4 CPU cycles. As soon as you have L1 caches misses, the memory load/stores are at more costly that that operation (setting a bit in a word).

That should be benched, but I doubt you will see any penalty for using `std::vector<bool>` instead of `std::vector<int>`